include/acl_types.h

```diff
@@ -337,6 +347,13 @@ typedef struct host_pipe_struct {
   int protocol = -1; // avalon_streaming = 0, avalon_streaming_uses_ready = 1
                      // avalon_mm = 2, avalon_mm_uses_ready = 3
 
+  // Number of sideband signals. Exclude data port.
+  unsigned num_side_band_signals = 0;
+
+  // Sideband signals map
+  // <offset, info>, sorted by offset
+  std::map<unsigned, sideband_signal_t> side_band_signals_map;
```

Comment:
Thanks Sophie for the comment.

I think using `std::set` instead of `std::map` requires me to define a '<' operator of the `sideband_signal_t` struct so I used `std::map` for convenient. Set and Map should have similar performance in this sense. In the initial design, the map doesn't contain the data port information so I wanted to sort the offset to find the data port size. 

Now, I kept the sorted data structure to read&write different part of the pointer in ascending order based on the offset value. I am not 100% sure if read/write in order is neccessary?

If we don't care of the order, I can change it to `unordered_set` all at once.

Reply:
Within the same data packet I don't think the ordering of pointer read/write should matter, it would be the same if you write to empty first then sop&eop or if you write sop&eop first then empty. As long as you ack everything altogether at the same time I think it should be fine.

Reply:
Thanks Sophie. Since we don't care about the order and there is no performance requirement for lookup, I am going to use a `vector` instead. I withdraw my earlier thought of using `unordered_set`, since it's unnecessary to use a hash table and creating a custom hash function for the custom struct.